tcms/report/views.py

```diff
@@ -81,71 +82,83 @@ def overview(request, product_id, template_name='report/overview.html'):
 
 class ProductVersionReport(TemplateView, ProductVersionReportData):
     template_name = 'report/version.html'
+    product = None
+    selected_version = None
 
-    def get(self, request, product_id):
+    def get(self, request, product_id):  # pylint: disable=arguments-differ
         try:
-            self.product = Product.objects.only('name').get(pk=product_id)
-
+            self.__class__.product = Product.objects.only('name').get(pk=product_id)
```

Comment:
Why `self.__class__.product` instead of `self.product` ? You can access class based attributes via the self object as well.